### PR TITLE
Add suport for CSS.

### LINKED ISSUE #23: Support for CSS
This would really help me. Just have the task do the exact same thing, but for CSS... If I have time, I'll implement this myself and put a pull request on this issue.

### PR DESCRIPTION
**Pull request by ruiaraujo**
Although the initial goal was to add support for css files, I ended up making it a bit more generic and customizable through a types array.

The CDN data was changed to support this. The current format is:

  ```javascript
  {
    'angular-material': {
      versions: ['0.7.1', '0.7.0'],
      js: function (version) {
        return '//my.own.cdn/libs/angular-material/' + version + '/angular-material.min.js';
      },
      css: function (version) {
        return '//my.own.cdn/libs/angular-material/' + version + '/angular-material.min.css';
      }
    }
  }
  ```

For each type, there should a function or a string in each CDN item. If it's not there, the 'url' is used as a fallback.
I can also submit PR for google-cdn-data and grunt-google-cdn if this goes through.

Fixes #23

### CODE REVIEW

**Comment by passy** [comment]
Sorry that this is still open. I'm a bit swamped at the moment.

File: googlecdn.js [hunk 1]
```diff
@@ -31,6 +31,26 @@ function isFunction(fn) {
...
+  var urlType = item[type];
+  if (urlType) {
+    return (isFunction(urlType)) ? urlType(version) : urlType;
+  }
+  if (item.url) {
+    return (isFunction(item.url)) ? item.url(version) : item.url;
+  }
+}
+
+function getArray(types) {
```
Comment (passy) [inline googlecdn.js]: Could you check for an existing function/module for this? Seems more node-y to reuse something for this instead.

File: googlecdn.js [hunk 2]
```diff
@@ -31,6 +31,26 @@ function isFunction(fn) {
   return typeof(fn) === 'function';
 }
 
+function getUrl(item, type, version) {
```
Comment (passy) [inline googlecdn.js]: I find this function a bit confusing. The name doesn't really describe what it does and it's only used in one place. Any other way to do this?

File: googlecdn.js [hunk 3]
```diff
@@ -100,16 +135,25 @@ module.exports = function cdnify(content, bowerJson, options, callback) {
...
+    }
+
     var replacementInfo = [];
 
     replacements.forEach(function (replacement) {
       if (replacement) {
-        content = content.replace(replacement.fromRegex, replacement.to);
-        debug('Replaced %s with %s', replacement.fromRegex, replacement.to);
-        replacementInfo.push(replacement);
+        if ( Array.isArray(replacement) ){
```
Comment (passy) [inline googlecdn.js]: Could you use the existing white space style here?

File: test.js [hunk 4]
```diff
@@ -177,7 +185,7 @@ describe('google-cdn', function () {
     });
   });
 
-  it('should print jquery-ui replacement information', function(cb){
+  it('should print jquery-ui replacement information', function (cb) {
```
Comment (passy) [inline test.js]: Thanks for fixing this! :+1:

**Comment by ruiaraujo** [comment]
I am currently a bit too busy to improve on this PR.

I will take a look later when I can.
I also want to take a look on how to handle multiple files per dependency with the same file type which is a requirements for certain angular libs out there.

If you have any suggestions on how to tackle that, let me know.

**Comment by ruiaraujo** [comment]
@passy I have updated the PR with the fixes requested as much as possible.

**Comment by douggr** [comment]
:+1: to merge

**Comment by passy** [comment]
Gosh, this is embarrassing. Sorry, this dropped off my radar entirely. I'll see what other steps are necessary to cut a new release tomorrow. Apologies for that ridiculously long delay.

**Comment by douggr** [comment]
No problem :) Thank you very much.

**Comment by ruiaraujo** [comment]
Better late than never.
Thanks @passy.

**Pull request merged**